File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/DataLakeMetadata.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Metadata of the dataLake
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=DataLakeMetadata.Builder.class)

public final class DataLakeMetadata  {
    @Deprecated
    @java.beans.ConstructorProperties({"auditLogId", "logGroupId", "isAuditEnabled", "auditRetentionPeriod"})
    public DataLakeMetadata(String auditLogId, String logGroupId, Boolean isAuditEnabled, Integer auditRetentionPeriod) {
        super();
        this.auditLogId = auditLogId;
        this.logGroupId = logGroupId;
        this.isAuditEnabled = isAuditEnabled;
        this.auditRetentionPeriod = auditRetentionPeriod;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Log object Id for DataLake logs
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("auditLogId")
        private String auditLogId;

                /**
         * Log object Id for DataLake logs
         * @param auditLogId the value to set
         * @return this builder
         **/
        

        public Builder auditLogId(String auditLogId) {
        this.auditLogId = auditLogId;
        return this;
        }
            /**
     * log groupId for the DataLake
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("logGroupId")
        private String logGroupId;

                /**
         * log groupId for the DataLake
         * @param logGroupId the value to set
         * @return this builder
         **/
        

        public Builder logGroupId(String logGroupId) {
        this.logGroupId = logGroupId;
        return this;
        }
            /**
     * is Audit enabled for the DataLake
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("isAuditEnabled")
        private Boolean isAuditEnabled;

                /**
         * is Audit enabled for the DataLake
         * @param isAuditEnabled the value to set
         * @return this builder
         **/
        

        public Builder isAuditEnabled(Boolean isAuditEnabled) {
        this.isAuditEnabled = isAuditEnabled;
        return this;
        }
            /**
     * DataLake Retention period for audit logs
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("auditRetentionPeriod")
        private Integer auditRetentionPeriod;

                /**
         * DataLake Retention period for audit logs
         * @param auditRetentionPeriod the value to set
         * @return this builder
         **/
        

        public Builder auditRetentionPeriod(Integer auditRetentionPeriod) {
        this.auditRetentionPeriod = auditRetentionPeriod;
        return this;
        }


        public DataLakeMetadata build() {
            DataLakeMetadata model = new DataLakeMetadata(this.auditLogId
                    , this.logGroupId
                    , this.isAuditEnabled
                    , this.auditRetentionPeriod);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(DataLakeMetadata model) {
                this.auditLogId(model.getAuditLogId());
    this.logGroupId(model.getLogGroupId());
    this.isAuditEnabled(model.getIsAuditEnabled());
    this.auditRetentionPeriod(model.getAuditRetentionPeriod());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Log object Id for DataLake logs
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("auditLogId")
    private final String auditLogId;

        /**
     * Log object Id for DataLake logs
     * @return the value
     **/
    
    public String getAuditLogId() {
        return auditLogId;
    }


        /**
     * log groupId for the DataLake
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("logGroupId")
    private final String logGroupId;

        /**
     * log groupId for the DataLake
     * @return the value
     **/
    
    public String getLogGroupId() {
        return logGroupId;
    }


        /**
     * is Audit enabled for the DataLake
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isAuditEnabled")
    private final Boolean isAuditEnabled;

        /**
     * is Audit enabled for the DataLake
     * @return the value
     **/
    
    public Boolean getIsAuditEnabled() {
        return isAuditEnabled;
    }


        /**
     * DataLake Retention period for audit logs
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("auditRetentionPeriod")
    private final Integer auditRetentionPeriod;

        /**
     * DataLake Retention period for audit logs
     * @return the value
     **/
    
    public Integer getAuditRetentionPeriod() {
        return auditRetentionPeriod;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("DataLakeMetadata(");
        sb.append("auditLogId=").append(String.valueOf(this.auditLogId));
        sb.append(", logGroupId=").append(String.valueOf(this.logGroupId));
        sb.append(", isAuditEnabled=").append(String.valueOf(this.isAuditEnabled));
        sb.append(", auditRetentionPeriod=").append(String.valueOf(this.auditRetentionPeriod));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof DataLakeMetadata)) {
            return false;
        }

        DataLakeMetadata other = (DataLakeMetadata) o;
        return java.util.Objects.equals(this.auditLogId, other.auditLogId) &&
            java.util.Objects.equals(this.logGroupId, other.logGroupId) &&
            java.util.Objects.equals(this.isAuditEnabled, other.isAuditEnabled) &&
            java.util.Objects.equals(this.auditRetentionPeriod, other.auditRetentionPeriod);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.auditLogId == null ? 43 : this.auditLogId.hashCode());
        result = (result * PRIME) + (this.logGroupId == null ? 43 : this.logGroupId.hashCode());
        result = (result * PRIME) + (this.isAuditEnabled == null ? 43 : this.isAuditEnabled.hashCode());
        result = (result * PRIME) + (this.auditRetentionPeriod == null ? 43 : this.auditRetentionPeriod.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/data_lake_metadata.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class DataLakeMetadata(object):
    """
    Metadata of the dataLake
    """

    def __init__(self, **kwargs):
        """
        Initializes a new DataLakeMetadata object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param audit_log_id:
            The value to assign to the audit_log_id property of this DataLakeMetadata.
        :type audit_log_id: str

        :param log_group_id:
            The value to assign to the log_group_id property of this DataLakeMetadata.
        :type log_group_id: str

        :param is_audit_enabled:
            The value to assign to the is_audit_enabled property of this DataLakeMetadata.
        :type is_audit_enabled: bool

        :param audit_retention_period:
            The value to assign to the audit_retention_period property of this DataLakeMetadata.
        :type audit_retention_period: int

        """
        self.swagger_types = {
            'audit_log_id': 'str',
            'log_group_id': 'str',
            'is_audit_enabled': 'bool',
            'audit_retention_period': 'int'
        }

        self.attribute_map = {
            'audit_log_id': 'auditLogId',
            'log_group_id': 'logGroupId',
            'is_audit_enabled': 'isAuditEnabled',
            'audit_retention_period': 'auditRetentionPeriod'
        }

        self._audit_log_id = None
        self._log_group_id = None
        self._is_audit_enabled = None
        self._audit_retention_period = None

    @property
    def audit_log_id(self):
        """
        Gets the audit_log_id of this DataLakeMetadata.
        Log object Id for DataLake logs


        :return: The audit_log_id of this DataLakeMetadata.
        :rtype: str
        """
        return self._audit_log_id

    @audit_log_id.setter
    def audit_log_id(self, audit_log_id):
        """
        Sets the audit_log_id of this DataLakeMetadata.
        Log object Id for DataLake logs


        :param audit_log_id: The audit_log_id of this DataLakeMetadata.
        :type: str
        """
        self._audit_log_id = audit_log_id

    @property
    def log_group_id(self):
        """
        Gets the log_group_id of this DataLakeMetadata.
        log groupId for the DataLake


        :return: The log_group_id of this DataLakeMetadata.
        :rtype: str
        """
        return self._log_group_id

    @log_group_id.setter
    def log_group_id(self, log_group_id):
        """
        Sets the log_group_id of this DataLakeMetadata.
        log groupId for the DataLake


        :param log_group_id: The log_group_id of this DataLakeMetadata.
        :type: str
        """
        self._log_group_id = log_group_id

    @property
    def is_audit_enabled(self):
        """
        Gets the is_audit_enabled of this DataLakeMetadata.
        is Audit enabled for the DataLake


        :return: The is_audit_enabled of this DataLakeMetadata.
        :rtype: bool
        """
        return self._is_audit_enabled

    @is_audit_enabled.setter
    def is_audit_enabled(self, is_audit_enabled):
        """
        Sets the is_audit_enabled of this DataLakeMetadata.
        is Audit enabled for the DataLake


        :param is_audit_enabled: The is_audit_enabled of this DataLakeMetadata.
        :type: bool
        """
        self._is_audit_enabled = is_audit_enabled

    @property
    def audit_retention_period(self):
        """
        Gets the audit_retention_period of this DataLakeMetadata.
        DataLake Retention period for audit logs


        :return: The audit_retention_period of this DataLakeMetadata.
        :rtype: int
        """
        return self._audit_retention_period

    @audit_retention_period.setter
    def audit_retention_period(self, audit_retention_period):
        """
        Sets the audit_retention_period of this DataLakeMetadata.
        DataLake Retention period for audit logs


        :param audit_retention_period: The audit_retention_period of this DataLakeMetadata.
        :type: int
        """
        self._audit_retention_period = audit_retention_period

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
